test/OpenTelemetry.Instrumentation.AspNet.Tests/HttpInListenerTests.cs

```diff
@@ -336,6 +336,108 @@ public void Dispose()
             Assert.Equal(HttpContext.Current.Request.UserAgent, span.GetTagValue(SemanticConventions.AttributeHttpUserAgent) as string);
         }
 
+        [Theory]
+        [InlineData(SamplingDecision.Drop)]
+        [InlineData(SamplingDecision.RecordOnly)]
+        [InlineData(SamplingDecision.RecordAndSample)]
+
+        public void ExtractContextIrrespectiveOfSamplingDecision(SamplingDecision samplingDecision)
+        {
+            HttpContext.Current = new HttpContext(
+                new HttpRequest(string.Empty, "http://localhost/", string.Empty)
+                {
+                    RequestContext = new RequestContext()
+                    {
+                        RouteData = new RouteData(),
+                    },
+                },
+                new HttpResponse(new StringWriter()));
+
+            bool isPropagatorCalled = false;
+            var propagator = new Mock<TextMapPropagator>();
+            propagator.Setup(m => m.Extract(It.IsAny<PropagationContext>(), It.IsAny<HttpRequest>(), It.IsAny<Func<HttpRequest, string, IEnumerable<string>>>()))
+                .Returns(() =>
+                {
+                    isPropagatorCalled = true;
+                    return default(PropagationContext);
+                });
+
+            var activity = new Activity(HttpInListener.ActivityOperationName);
+
+            var activityProcessor = new Mock<BaseProcessor<Activity>>();
+            Sdk.SetDefaultTextMapPropagator(propagator.Object);
+            using (var openTelemetry = Sdk.CreateTracerProviderBuilder()
+                .SetSampler(new TestSampler(samplingDecision))
+                .AddAspNetInstrumentation()
+                .AddProcessor(activityProcessor.Object).Build())
+            {
+                activity.Start();
+
+                using (var inMemoryEventListener = new InMemoryEventListener(AspNetInstrumentationEventSource.Log))
+                {
+                    this.fakeAspNetDiagnosticSource.Write("Start", null);
+                }
+
+                this.fakeAspNetDiagnosticSource.Write("Stop", null);
+                activity.Stop();
+            }
+
+            Assert.True(isPropagatorCalled);
```

Comment:
Compared to PR 1934, this is only checking if propagator is called or not. That alone is not sufficient, as the actual thing to be validated is if the Baggage.Current and Activity.Current is setup based on incoming headers.

(This can be added as a follow up.)